source/_posts/2019-07-15-complete-flows-partial-models.markdown

```diff
@@ -0,0 +1,166 @@
+---
+layout: post
+author: rafa
+title: "(Complete, Flows) -> (Partial) -> (Models)"
+date: 2019-07-15 19:55:56 +0200
+comments: true
+categories: 
+---
+
+Most apps these days have a sequence of screens that gather information from the user, like a registration flow, a form of some kind. The data from each step is typically combined into a single data structure.
+For example, let's say we want the name, age, and the password to authenticate the user.
+<!--more-->
+One way to model it is by using the following data structure:
+
+```swift
+struct FormData {
+    let name: String
+    let age: Int
+    let password: String
+}
+```
+
+One issue we are going to come about is that our model is strict, it needs all the values at once, whereas users will supply each value at a time. First they will type in their name, then their age, and so on.
+Wrapping up the fields in `Optional`, may loosen its strictness.
```

Comment:
It's super obvious, but maybe having a note on why loosening up the strictness of the structure is a bad thing?